src/components/MultiSelect/MultiSelect.stories.mdx

```diff
@@ -0,0 +1,322 @@
+
+Resources:
+
+- [downshift.js documentation](https://www.downshift-js.com/)
+- [Reservoir Checkbox Component](http://localhost:6006/?path=/docs/components-form-elements-checkbox--checkbox#accessibility)
+- [W3C WAI Listbox Example](https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/listbox/listbox.html)
+- [W3C WAI on using ARIA role=dialog](https://www.w3.org/WAI/GL/wiki/Using_ARIA_role%3Ddialog_to_implement_a_modal_dialog_box)
+
+## Using the items prop
+
+The options rendered in the MultiSelect are passed into the component's items prop.
+The expected data structure is an array of objects, in the following format:
+
+```tsx
+<MultiSelect
+  id="group1"
+  label="Subject Specialities"
+  items={[
+    {
+      id: "art",
+      name: "Art",
+    },
+    {
+      id: "architecture",
+      name: "Architecture",
+    },
+    {
+      id: "art_history",
+      name: "Art History",
+    },
+  ]}
+/>
+```
+
+You can also add a second level of child items, in the following format:
+
+```tsx
+<MultiSelect
+  id="group1"
+  label="Subject Specialities"
+  items={[
+    {
+      id: "art",
+      name: "Art",
+    },
+    {
+      id: "parent_option",
+      name: "Parent option",
+      children: [
+        {
+          id: "child_1",
+          name: "Child 1",
+        },
+        {
+          id: "child_2",
+          name: "Child 2",
+        },
+      ],
+    },
+  ]}
+/>
+```
+
+## Controlling state using selectedItems and onChange prop
+
+While the component does not store the selectedItems state internally, it expects
+a certain state shape to be used for calculating the selected item count. This
+state shape is a "normalized" pattern, and is basically an object of objects,
+keyed by the multiselect id.
+
+```json
+{
+  "subjects": {
+    "items": ["art", "architecture"]
+  },
+  "locations": {
+    "items": ["sasb"]
+  }
+}
+```
+
+This is helpful for when several MultiSelect components are rendered inside a FilterBar component,
+you can use the MultiSelect id to get items for a specific slice of the state.
+
+```jsx
+const subjectsSelectedItemssOnly = selectedItems["subjects"].items;
+const locationsSelectedItemssOnly = selectedItems["locations"].items;
+```
+
+It's also helpful if a consuming app is pushing state into the browser's url.
+This example uses NextJS's router push method to add query parameters to url.
+
+```jsx
+const multiSelectId = "subjects";
+// This wouldn't be a const in real world code, but for display purposes here
+// to show the state object more clearly.
+const mockStateObject = {
+  subjects: {
+    items: ["art", "architecture"],
+  },
+  locations: {
+    items: ["sasb"],
+  },
+};
+
+// Will push ?subjects=art,architecture into browser url.
+router.push({
+  query: {
+    [multiSelectId]: selectedItems[multiSelectId].items.join(" "),
+  },
+});
+```
```

Comment:
This would be great in a separate section at the end for "MultiSelect NextJS routing implementation" heading.

Reply:
fixed